### PR TITLE
[RPD-182] Refactor analytics opt-out/in into core

### PR DESCRIPTION
**Pull request by swells2020**
This pull request refactors the functionality of the `analytics` CLI module into `matcha_ml.cli.cli.py` and `matcha_ml.core.core.py` by moving the CLI logic into the `cli.py` module and the functionality that alters the global config file into `core.py`.

The tests have also been split, with `test_analytics.py` now only testing the analytics components of the CLI module. Three additional tests have been included; testing the `--opt-in` option, the `--opt-out` option, and the `--help` default. The tests of the core functionality have been moved to `test_core.py` and are unchanged, other than updating imports.

## Checklist

Please ensure you have done the following:

* [X] I have read the [CONTRIBUTING](https://github.com/fuzzylabs/matcha/blob/main/CONTRIBUTING.md) guide.
* [ ] I have updated the documentation if required.
* [X] I have added tests which cover my changes.

## Type of change

Tick all those that apply:

* [ ] Bug Fix (non-breaking change, fixing an issue)
* [ ] New feature (non-breaking change to add functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [X] Other (add details above)

### CODE REVIEW

File: tests/test_cli/test_analytics.py [hunk 1]
```diff
@@ -1,132 +1,94 @@
...
+        "Enable or disable the collection of anonymous usage data (enabled by default)."
+        in result.stdout
     )
 
-    expected_configuration["analytics_opt_out"] = True
 
-    # Check if config file is not present
-    assert not os.path.exists(config_file_path)
+def test_cli_analytics_command_default_help(runner: CliRunner) -> None:
+    """Test cli for analytics command help.
```
Comment (Christopher-Norman) [inline tests/test_cli/test_analytics.py:31]: Could this be more descriptive?

Comment (swells2020) [reply tests/test_cli/test_analytics.py:31]: Absolutely, I'll reword it to 'test_cli_analytics_command_defaults_to_help'.

File: src/matcha_ml/cli/cli.py [hunk 2]
```diff
@@ -4,7 +4,6 @@
 import typer
 
 from matcha_ml import __version__
-from matcha_ml.cli import analytics
```
[the remark this answers is not in the document]
Comment (swells2020) [reply src/matcha_ml/cli/cli.py:7]: We are. The functions are being moved to `core.py`, similar to the way Oscar implemented `get`, and the cli logic has been kept within `cli.py`. This will make the API easier to implement in the future.

**Comment by swells2020** [comment]
> Looking good, just a couple of comments/questions

Thanks Chris 🙏 made some changes as suggested, hope this works.
